### PR TITLE
Bump version to v0.3.1

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

**Review by BurntSushi — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,40 @@
...
+- \[`pycodestyle`\] Fix E301 not triggering on decorated methods. ([#10117](https://github.com/astral-sh/ruff/pull/10117))
+- \[`pycodestyle`\] Respect `isort` settings in blank line rules (`E3*`) ([#10096](https://github.com/astral-sh/ruff/pull/10096))
+- \[`pycodestyle`\] Make blank lines in typing stub files optional (`E3*`) ([#10098](https://github.com/astral-sh/ruff/pull/10098))
+- \[`pylint`\] Implement `singledispatch-method` (`E1519`) ([#10140](https://github.com/astral-sh/ruff/pull/10140))
+- \[`pylint`\] Implement `useless-exception-statement` (`W0133`) ([#10176](https://github.com/astral-sh/ruff/pull/10176))
+
+### Rule changes
+
+- \[`flake8-debugger`\] Check for use of `debugpy` and `ptvsd` debug modules (#10177) ([#10194](https://github.com/astral-sh/ruff/pull/10194))
+- \[`pyupgrade`\] Generate diagnostic for all valid f-string conversions regardless of line-length (`UP032`) ([#10238](https://github.com/astral-sh/ruff/pull/10238))
```
Comment (AlexWaygood) [inline CHANGELOG.md:16]: nit

```suggestion
- \[`pyupgrade`\] Generate diagnostic for all valid f-string conversions regardless of line length (`UP032`) ([#10238](https://github.com/astral-sh/ruff/pull/10238))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+- \[`pep8_naming`\] Add fixes `N804` and `N805` ([#10215](https://github.com/astral-sh/ruff/pull/10215))
```
Comment (AlexWaygood) [inline CHANGELOG.md:17]:
```suggestion
- \[`pep8_naming`\] Add fixes for `N804` and `N805` ([#10215](https://github.com/astral-sh/ruff/pull/10215))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,40 @@
...
+
+### CLI
+
+- Colorize the output of `ruff format --diff` ([#10110](https://github.com/astral-sh/ruff/pull/10110))
+- Make `--config` and `--isolated` global flags ([#10150](https://github.com/astral-sh/ruff/pull/10150))
+- Move shell expansion into `--config` lookup ([#10219](https://github.com/astral-sh/ruff/pull/10219))
```
Comment (AlexWaygood) [inline CHANGELOG.md:23]:
```suggestion
- Correctly expand tildes and environment variables in paths passed to `--config` ([#10219](https://github.com/astral-sh/ruff/pull/10219))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,40 @@
...
+### Configuration
+
+- Accept a PEP 440 version specifier for `required-version` ([#10216](https://github.com/astral-sh/ruff/pull/10216))
+- Implement isort's `default-section` setting ([#10149](https://github.com/astral-sh/ruff/pull/10149))
+
+### Bug fixes
+
+- Remove trailing space from `CapWords` message ([#10220](https://github.com/astral-sh/ruff/pull/10220))
+- Respect external codes in file-level exemptions ([#10203](https://github.com/astral-sh/ruff/pull/10203))
+- \[`flake8-raise`\] Avoid false-positives for parens-on-raise with `futures.exception()` (`RSE102`) ([#10206](https://github.com/astral-sh/ruff/pull/10206))
```
Comment (AlexWaygood) [inline CHANGELOG.md:34]:
```suggestion
- \[`flake8-raise`\] Avoid false-positives for parens-on-raise with `future.exception()` (`RSE102`) ([#10206](https://github.com/astral-sh/ruff/pull/10206))
```

**Review by AlexWaygood — COMMENTED** [review]
A few nits ;)

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**